includes/gateways/functions.php

```diff
@@ -345,19 +356,30 @@ function give_count_sales_by_gateway( $gateway_id = 'paypal', $status = 'publish
 /**
  * Returns a ordered list of all available gateways.
  *
- * @unlreased add $version param
+ * @unreleased add $version param
  * @since 1.4.5
  *
  * @param array $gateways List of payment gateways
  * @param int|null $version
  * @return array $gateways All the available gateways
  */
-function give_get_ordered_payment_gateways($gateways, $version = 2)
+function give_get_ordered_payment_gateways($gateways, $version = null)
 {
-    $suffix = $version === 3 ? '_v3' : '';
-
 	// Get gateways setting.
-    $gateways_setting = $_POST['gateways' . $suffix] ?? give_get_option('gateways' . $suffix);
+    $gateways_setting = [];
+    if (!$version || $version === 2) {
+        $gateways_setting = array_merge(
+            $gateways_setting,
+            $_POST['gateways'] ?? (array)give_get_option('gateways', [])
+        );
+    }
+
+    if (!$version || $version === 3) {
+        $gateways_setting = array_merge(
+            $gateways_setting,
+            $_POST['gateways_v3'] ?? (array)give_get_option('gateways_v3', [])
+        );
+    }
```

Comment:
Same comment here about the `$_POST` request to be resolved as array

Reply:
08fbab2fcdd384ae5c6550d00056edc5ee3ef7dc